web/src/components/Settings/WorkspaceSection.tsx

```diff
@@ -50,6 +53,15 @@ const WorkspaceSection = () => {
     toast.success(t("message.update-succeed"));
   };
 
+  useEffect(() => {
+    fetchIdentityProviderList();
+  }, []);
+
+  const fetchIdentityProviderList = async () => {
+    const { identityProviders } = await identityProviderServiceClient.listIdentityProviders({});
```

Comment:
I've seen the code use a `useasync` pattern. Is this applicable here?

Reply:
> I've seen the code use a `useasync` pattern. Is this applicable here?

I have already copied all this from the SSO section it is using the exact same code